src/SixLabors.Fonts/Tables/Cff/Cff1Parser.cs

```diff
@@ -0,0 +1,1256 @@
+
+            // The client program will contain an array of standard strings with
+            // nStdStrings elements.
+            // Thus, the standard strings take SIDs in the
+            // range 0 to(nStdStrings –1).
+
+            // The first string in the String INDEX
+            // corresponds to the SID whose value is equal to nStdStrings, the
+            // first non - standard string, and so on.
+
+            // When the client needs to
+            // determine the string that corresponds to a particular SID it
+            // performs the following: test if SID is in standard range then
+            // fetch from internal table,
+            // otherwise, fetch string from the String
+            // INDEX using a value of(SID – nStdStrings) as the index.
+            if (!this.TryReadIndexDataOffsets(reader, out CffIndexOffset[]? offsets))
+            {
+                return Array.Empty<string>();
+            }
+
+            string[] stringIndex = new string[offsets.Length];
+
+            // Allow reusing the same buffer for shorter reads.
+            using var buffer = new Buffer<byte>(512);
+            Span<byte> bufferSpan = buffer.GetSpan();
+
+            for (int i = 0; i < offsets.Length; ++i)
+            {
+                int length = offsets[i].Length;
+                if (length < bufferSpan.Length)
+                {
+                    Span<byte> slice = bufferSpan.Slice(0, length);
+                    int actualRead = reader.BaseStream.Read(slice);
+#if DEBUG
+                    if (actualRead != length)
+                    {
+                        throw new NotSupportedException();
+                    }
+#endif
+                    stringIndex[i] = Iso88591.GetString(slice);
+                }
+                else
+                {
+                    stringIndex[i] = Iso88591.GetString(reader.ReadBytes(length), 0, length);
+                }
+            }
+
+            return stringIndex;
+        }
+
+        private string GetSid(int index, string[] stringIndex)
+        {
+            if (index >= 0 && index <= CffStandardStrings.Count - 1)
+            {
+                // Use standard name
+                return CffStandardStrings.GetName(index);
+            }
+
+            if (index - CffStandardStrings.Count < stringIndex.Length)
+            {
+                return stringIndex[index - CffStandardStrings.Count];
+            }
+
+            // Technically this maps to .notdef, but PDFBox uses this
+            return "SID" + index;
+        }
+
+        private CffTopDictionary ResolveTopDictInfo(List<CffDataDicEntry> entries, string[] stringIndex)
+        {
+            // TODO: Is CID mandatory?
+            CffTopDictionary metrics = new();
+            foreach (CffDataDicEntry entry in entries)
+            {
+                switch (entry.Operator.Name)
+                {
+                    default:
+#if DEBUG
```

Comment:
I dont like those `#if DEBUG` statements much.

Maybe we should introduce a proper logger for such cases? I really like [serilog](https://serilog.net) for such things. I am a big fan of structured logging. I can really help finding bugs/issues.